interfaces/builtin/kubernetes_support.go

```diff
@@ -35,65 +47,226 @@ const kubernetesSupportBaseDeclarationSlots = `
+# Allow running as the kubeproxy service
+
+# managing our own cgroup
+/sys/fs/cgroup/*/kube-proxy/{,**} rw,
 
 # Allow reading the state of modules kubernetes needs
+/sys/module/libcrc32c/initstate r,
 /sys/module/llc/initstate r,
 /sys/module/stp/initstate r,
+/sys/module/ip_vs/initstate r,
+/sys/module/ip_vs_rr/initstate r,
+/sys/module/ip_vs_sh/initstate r,
+/sys/module/ip_vs_wrr/initstate r,
+`
 
-# Allow listing kernel modules. Note, seccomp blocks module loading syscalls
-/sys/module/apparmor/parameters/enabled r,
-/bin/kmod ixr,
-/etc/modprobe.d/{,**} r,
+var kubernetesSupportConnectedPlugKmodKubeProxy = []string{
+	`ip_vs_rr`,
+	`ip_vs_sh`,
+	`ip_vs_wrr`,
+	`libcrc32c`,
+	`llc`,
+	`stp`,
+}
 
-# Allow ptracing Docker containers
-ptrace (read, trace) peer=docker-default,
-ptrace (read, trace) peer=snap.docker.dockerd,
+type kubernetesSupportInterface struct{}
 
-# Should we have a 'privileged' mode for kubernetes like we do with
-# docker-support? Right now kubernetes needs this which allows it to control
-# all processes on the system and on <4.8 kernels, escape confinement.
-ptrace (read, trace) peer=unconfined,
-`
+func (iface *kubernetesSupportInterface) Name() string {
+	return "kubernetes-support"
+}
 
-var kubernetesSupportConnectedPlugKmod = []string{`llc`, `stp`}
+func (iface *kubernetesSupportInterface) StaticInfo() interfaces.StaticInfo {
+	return interfaces.StaticInfo{
+		Summary:              kubernetesSupportSummary,
+		BaseDeclarationPlugs: kubernetesSupportBaseDeclarationPlugs,
+		BaseDeclarationSlots: kubernetesSupportBaseDeclarationSlots,
+		ImplicitOnCore:       true,
+		ImplicitOnClassic:    true,
+	}
+}
+
+func k8sFlavor(plug *interfaces.ConnectedPlug) string {
+	var flavor string
+	_ = plug.Attr("flavor", &flavor)
+	return flavor
+}
+
+func (iface *kubernetesSupportInterface) AppArmorConnectedPlug(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	snippet := kubernetesSupportConnectedPlugAppArmorCommon
+	systemd_run_extra := ""
+
+	switch flavor := k8sFlavor(plug); flavor {
+	case "kubelet":
+		systemd_run_extra = kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun
+		snippet += kubernetesSupportConnectedPlugAppArmorKubelet
+		spec.UsesPtraceTrace()
+	case "kubeproxy":
+		snippet += kubernetesSupportConnectedPlugAppArmorKubeproxy
+	default:
+		systemd_run_extra = kubernetesSupportConnectedPlugAppArmorKubeletSystemdRun
+		snippet += kubernetesSupportConnectedPlugAppArmorKubelet
+		snippet += kubernetesSupportConnectedPlugAppArmorKubeproxy
+		spec.UsesPtraceTrace()
+	}
+
+	old := "###KUBERNETES_SUPPORT_SYSTEMD_RUN###"
+	spec.AddSnippet(strings.Replace(snippet, old, systemd_run_extra, -1))
+	return nil
+}
+
+func (iface *kubernetesSupportInterface) SecCompConnectedPlug(spec *seccomp.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	flavor := k8sFlavor(plug)
+	if flavor == "kubelet" || flavor == "" {
```

Comment:
Is `flavor. == ""` a default that really means `kubelet`? If so could that knowledge be folded into `k8sFlavor`?

Reply:
Default means "all policy", ie, *both* kubelet and kubeproxy and any future policy. The kubelet and kubeproxy flavors are for when you want to limit kubelet and (particularly) kubeproxy to not have all of the policy of the other.